### PR TITLE
fix: display required indicator only for non-empty labels and Improve test structure

### PR DESCRIPTION
**Pull request by amitbadala**
## Summary of change

Fixes and feedbacks from [this PR](https://github.com/supertokens/supertokens-auth-react/pull/760)

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.
-   [ ] If I added a new recipe, I also added the recipe entry point into the `rollup.config.mjs`

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: lib/ts/recipe/emailpassword/components/library/label.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export default function Label({ value, showIsRequired }: LabelProps): JSX.Elemen
     return (
         <div data-supertokens="label">
             {t(value)}
-            {showIsRequired && " *"}
+            {showIsRequired && value !== "" && " *"}
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailpassword/components/library/label.tsx:29]: what if the label has spaces like `"  "`? Shouldn't we trim it? Or is it already trimmed by the time it comes here?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/emailpassword/components/library/label.tsx (hunk 1 above)
Comment (amitbadala) [reply lib/ts/recipe/emailpassword/components/library/label.tsx]: added a new commit for same

**Review by porcellus — CHANGES_REQUESTED** [review]

File: examples/for-tests/src/App.js [hunk 2]
```diff
@@ -747,22 +744,20 @@ function getSignInFormFields() {
...
+            return [
                 {
                     id: "email",
                     nonOptionalErrorMsg: "Please add email",
                 },
-            ],
-        };
+            ];
+        default:
+            return [];
```
[the remark this answers is not in the document]
Comment (amitbadala) [reply examples/for-tests/src/App.js:756]: agreed

**Review by porcellus — APPROVED** [review]

File: lib/ts/recipe/emailpassword/components/library/label.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ export default function Label({ value, showIsRequired }: LabelProps): JSX.Elemen
     return (
         <div data-supertokens="label">
             {t(value)}
-            {showIsRequired && " *"}
+            {showIsRequired && value && value.trim() !== "" && " *"}
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailpassword/components/library/label.tsx:29]: add a test for the trimming too

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Pull request merged**